### PR TITLE
Fix bug in DISTINCT

### LINKED ISSUE #49622: Unexpected "DISTINCT" result with empty string
+ ClickHouse Version

    `23.4.2.11`

+ DDL
    ```sql
    CREATE TABLE test
    (
        c1 String,
        c2 String,
        c3 String
    )
    ENGINE = ReplacingMergeTree
    ORDER BY (c1, c3);
    
    INSERT INTO test(c1, c2, c3) VALUES
        ('', '', '1'), ('', '', '2'),('v1', 'v2', '3'),('v1', 'v2', '4'),('v1', 'v2', '5');
    ```

```sql

SELECT c1, c2, c3 FROM test GROUP BY c1, c2, c3;
┌─c1─┬─c2─┬─c3─┐
│    │    │ 2  │
│ v1 │ v2 │ 4  │
│ v1 │ v2 │ 3  │
│    │    │ 1  │
│ v1 │ v2 │ 5  │
└────┴────┴────┘

SELECT DISTINCT c1, c2, c3 FROM test;
┌─c1─┬─c2─┬─c3─┐
│    │    │ 1  │
│ v1 │ v2 │ 3  │
│ v1 │ v2 │ 4  │
│ v1 │ v2 │ 5  │
└────┴────┴────┘

```
Why does the query result of this DISTINCT clause not contain rows where c3 is 2

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Turn off the optimization of DISTINCT in order by default, because it has a bug. This closes #49622.

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit d0400846f119da1a6c8fd3681338f499e3649c31 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49628/d0400846f119da1a6c8fd3681338f499e3649c31/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by devcrafter — APPROVED** [review]

File: tests/queries/0_stateless/02733_distinct.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    c2 String,
+    c3 String
+)
+ENGINE = ReplacingMergeTree
+ORDER BY (c1, c3);
+
+INSERT INTO test(c1, c2, c3) VALUES ('', '', '1'), ('', '', '2'),('v1', 'v2', '3'),('v1', 'v2', '4'),('v1', 'v2', '5');
+
+SELECT c1, c2, c3 FROM test GROUP BY c1, c2, c3 ORDER BY c1, c2, c3;
+SELECT DISTINCT c1, c2, c3 FROM test ORDER BY c1, c2, c3;
```
Comment (devcrafter) [inline tests/queries/0_stateless/02733_distinct.sql:14]: DISTINCT with ORDER BY works correctly, but DISTINCT w/o ORDER BY affected by the bug:
```suggestion
SELECT DISTINCT c1, c2, c3 FROM test;
```

**Comment by devcrafter** [comment]
@alexey-milovidov 
What was the point to merging it if there is the fix (https://github.com/ClickHouse/ClickHouse/pull/49636) ?

**Comment by alexey-milovidov** [comment]
We can quickly backport disabling the optimization while the proper fix is still being discussed.

**Pull request merged**